src/ControlSystem/Metafunctions.hpp

```diff
@@ -105,5 +112,23 @@ using interpolation_target_tags = tmpl::flatten<tmpl::transform<
             tmpl::_1,
             control_systems_with_measurement<tmpl::pin<ControlSystems>,
                                              tmpl::parent<tmpl::_1>>>>>>>;
+
+namespace detail {
+CREATE_GET_TYPE_ALIAS_OR_DEFAULT(component_being_mocked)
+
+template <typename Metavariables>
+using all_components = typename Metavariables::component_list;
+
+template <typename Metavariables>
+using all_not_mocked_components =
+    tmpl::transform<all_components<Metavariables>,
+                    get_component_being_mocked_or_default<tmpl::_1, tmpl::_1>>;
```

Comment:
@wthrowe Squashed in the changes because they were just file renames and adding one function in the SystemHelpers. I also changed this from `tmpl::bind<get_component_being_mocked_or_default_t, ...>` to `get_component_being_mocked_or_default<...>`